### PR TITLE
#217 Button icon toggle

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-design/issues/217 |
| Decisions | Add button icon toogle based on button-icon component to add internal value toggle logic. |
| Animated GIF | b |

### CODE REVIEW

File: vue/test/components/button-icon-toggle/button-icon-toggle.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const assert = require("assert");
```
Comment (gcandal) [inline vue/test/components/button-icon-toggle/button-icon-toggle.test.js:1]: Come on, you can do better :D

Comment (gcandal) [reply vue/test/components/button-icon-toggle/button-icon-toggle.test.js:1]: (more tests)

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 2]
```diff
@@ -0,0 +1,98 @@
+<template>
+    <button-icon
+        class="button-icon-toggle"
+        v-bind:icon="iconComputed"
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:4]: attrs

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        color: {
+            type: String,
+            default: "gray"
+        },
+        /**
+         * The color of the button when `value` is true.
+         */
+        colorSecondary: {
+            type: String,
+            default: null
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:46]: make up some default

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            mandatory: true
+        },
+        /**
+         * The name of the icon to appear in the middle of the button.
+         * This icon will show when `value` is false.
+         * If none is received, `icon` will be considered instead.
+         */
+        iconSecondary: {
+            type: String,
+            default: null
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:32]: define a default

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+         * @values true, false
+         */
+        value: {
+            type: Boolean,
+            default: false
+        }
+    },
+    data: function() {
+        return {
+            animating: false,
```
Comment (3rdvision) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:66]: delete

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+         * This icon will show when `value` is true.
+         */
+        icon: {
+            type: String,
+            mandatory: true
+        },
+        /**
+         * The name of the icon to appear in the middle of the button.
+         * This icon will show when `value` is false.
+         * If none is received, `icon` will be considered instead.
```
Comment (gcandal) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:28]: No longer true, right?

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        /**
+         * The color of the button when `value` is true.
+         */
+        colorSecondary: {
+            type: String,
+            default: "black"
+        },
+        /**
+         * The value of the button.
+         * @values true, false
```
Comment (gcandal) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:50]: No need for this, we know what a Boolean is :D

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            this.$emit("update:value", value);
+        }
+    },
+    computed: {
+        colorComputed() {
+            return this.valueData ? this.colorSecondary || this.color : this.color;
+        },
+        iconComputed() {
+            return this.valueData ? this.iconSecondary || this.icon : this.icon;
+        }
```
Comment (gcandal) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.vue:80]: You can assume both secondary are required or have a default, so the fallback is not needed.

**Review by gcandal — CHANGES_REQUESTED** [review]

**Comment by gcandal** [comment]
@3rdvision please substitute the PNG with a GIF. You might argue that going to the storybook is the same, but when you have 80 PRs to review this stuff matters.

**Review by gcandal — APPROVED** [review]

File: vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.stories.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+                this.valueData = value;
+            }
+        },
+        data: function() {
+            return {
+                valueData: this.value
+            };
+        },
+        template: `
+        <div>
```
Comment (joamag) [inline vue/components/ui/molecules/button-icon-toggle/button-icon-toggle.stories.js:75]: fix indentation

**Review by joamag — CHANGES_REQUESTED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**